models/User.js

```diff
@@ -61,13 +46,23 @@ UserSchema.statics.authenticate = function (userName, password, cb) {
 };
 
 UserSchema.statics.createUser = function (userName, password, cb) {
-  const user = new User({ userName, passwordHash: password });
 
-  user.save((err) => {
-    if (err) {
-      return cb(err);
-    }
-    cb(null, this);
+  bcrypt.genSalt(saltRounds, (err, salt) => {
```

Comment:
Hide this in a user instance method.  i.e. `user.setPassword(password, callback)`